crates/telio-dns/tests/nameserver.rs

```diff
@@ -152,14 +154,29 @@ impl WGClient {
             .decapsulate(None, &receiving_buffer[..bytes_read], &mut sending_buffer)
         {
             TunnResult::WriteToTunnelV4(response, _) => {
+                if test_type.is_ipv6() {
+                    panic!("Decapsulate into IPv4 for IPv6");
+                }
+
                 let ip_response = Ipv4Packet::new(response).expect("Failed to parse ip response");
                 let udp_response =
                     UdpPacket::new(ip_response.payload()).expect("Failed to parse udp response");
                 dns_parser::Packet::parse(udp_response.payload())
                     .expect("Failed to parse dns response");
             }
+            TunnResult::WriteToTunnelV6(response, _) => {
+                if test_type.is_ipv4() {
+                    panic!("Decapsulate into IPv6 for IPv4");
+                }
+
+                let ip_response = Ipv6Packet::new(response).expect("Failed to parse ip response");
+                let udp_response =
+                    UdpPacket::new(ip_response.payload()).expect("Failed to parse udp response");
+                dns_parser::Packet::parse(udp_response.payload())
+                    .expect("Failed to parse dns response");
+            }
             TunnResult::Err(e) => panic!("Decapsulate error: {:?}", e),
-            _ => println!("Unexpected TunnResult while receiving the dns response"),
+            _ => panic!("Unexpected TunnResult while receiving the dns response"),
```

Comment:
This panic seems dangerous, can we handle this in less dangerous way, e.g. `telio_log_err()` or similar. 

Also `TunnResult::Err(e)` case is probably dangerous in a similar manner. Can we handle it in some less destructive way? :thinking:

Reply:
Oh wait, I misread this and missed the fact that this is test code.

Reply:
It's in a test so should not be a problem